app/scripts/modules/core/src/pipeline/config/stages/executionWindows/ExecutionWindows.tsx

```diff
@@ -0,0 +1,409 @@
+  const defaultJitter: IJitter = {
+    minDelay: 0,
+    maxDelay: 600,
+    enabled: false,
+    skipManual: false,
+  };
+  const jitter = get(props.restrictedExecutionWindow, 'jitter', defaultJitter);
+  const whitelist = get(props.restrictedExecutionWindow, 'whitelist', []);
+  const days = get(props.restrictedExecutionWindow, 'days', []);
+  const [timelineWindows, setTimelineWindows] = useState(getTimelineWindows(whitelist));
+  const [enableCustomSkipWindowText, setEnableCustomSkipWindowText] = useState(!!props.skipWindowText);
+  const defaultSkipWindowText = DEFAULT_SKIP_WINDOW_TEXT;
+  const isWindowExpression =
+    typeof props.restrictExecutionDuringTimeWindow === 'string' &&
+    props.restrictExecutionDuringTimeWindow.includes('${');
+  const hours = TimePickerOptions.getHours();
+  const minutes = TimePickerOptions.getMinutes();
+  const dividers = hours.map(hour => ({
+    label: hour.label,
+    left: (hour.value / 24) * 100 + '%',
+  }));
+
+  useEffect(() => {
+    if (!props.restrictExecutionDuringTimeWindow) {
+      props.updateStageField({
+        restrictedExecutionWindow: {
+          whitelist: [],
+          days: [],
+          jitter: defaultJitter,
+        },
+      });
+      setTimelineWindows([]);
+    }
+  }, [props.restrictExecutionDuringTimeWindow]);
+
+  const jitterUpdated = (changes: Partial<IJitter>) => {
+    let newJitterValue: IJitter = { ...jitter };
+    extend(newJitterValue, changes);
+    if (newJitterValue.enabled) {
+      if (newJitterValue.minDelay >= 0 && newJitterValue.maxDelay <= newJitterValue.minDelay) {
+        newJitterValue.maxDelay = newJitterValue.minDelay + 1;
+      }
+    } else {
+      newJitterValue = defaultJitter;
+    }
+    const restrictedExecutionWindow = {
+      ...props.restrictedExecutionWindow,
+      jitter: newJitterValue,
+    };
+    props.updateStageField({ restrictedExecutionWindow });
+  };
+
+  const addExecutionWindow = (): void => {
+    const newExecutionWindow = {
+      startHour: 0,
+      startMin: 0,
+      endHour: 0,
+      endMin: 0,
+    };
+    const restrictedExecutionWindow = {
+      ...props.restrictedExecutionWindow,
+      whitelist: [...whitelist, newExecutionWindow],
+    };
+    props.updateStageField({ restrictedExecutionWindow });
+  };
+
+  const removeWindow = (index: number): void => {
+    const newWhitelist = [...whitelist];
+    newWhitelist.splice(index, 1);
+    const restrictedExecutionWindow = {
+      ...props.restrictedExecutionWindow,
+      whitelist: newWhitelist,
+    };
+    props.updateStageField({ restrictedExecutionWindow });
+  };
+
+  const windowDaysUpdated = (newDays: number[]): void => {
+    const restrictedExecutionWindow = {
+      ...props.restrictedExecutionWindow,
+      days: newDays,
+    };
+    props.updateStageField({ restrictedExecutionWindow });
+  };
+
+  const updateWhiteList = (changes: Partial<IWindow>, index: number) => {
```

Comment:
super minor: we are not capitalizing `List` in `whitelist` elsewhere.

Reply:
👍